ElectronClient/plugins/GotoAnything.jsx

```diff
@@ -191,8 +191,11 @@ class Dialog extends React.PureComponent {
 
 			if (this.state.query.indexOf('#') === 0) { // TAGS
 				listType = BaseModel.TYPE_TAG;
-				searchQuery = `*${this.state.query.split(' ')[0].substr(1).trim()}*`;
-				results = await Tag.searchAllWithNotes({ titlePattern: searchQuery });
+				searchQuery = this.state.query.split(' ')[0].substr(1).trim();
+				results = await Tag.search({ fullTitleRegex: `.*${searchQuery}.*` });
+				for (let i = 0; i < results.length; i++) {
+					results[i].title = Tag.getCachedFullTitle(results[i].id);
```

Comment:
Please don't mutate data, as it could cause hard to debug issues. I've noticed a few instances of this, so I'll mark them below. In this case, using `map()` to modify the array, and `Object.assign` to modify the objects might be best:
`results = results.map(tag => Object.assign({}, tag, { title: Tag.getCachedFullTitle(tag.id) })`